fineract-provider/src/main/java/org/apache/fineract/portfolio/delinquency/service/DelinquencyWritePlatformServiceImpl.java

```diff
@@ -0,0 +1,195 @@
+        DelinquencyRange delinquencyRange = updateOrCreateDelinquencyRange(data, changes);
+        return new CommandProcessingResultBuilder().withCommandId(command.commandId()).withEntityId(delinquencyRange.getId()).with(changes)
+                .build();
+    }
+
+    @Override
+    public CommandProcessingResult updateDelinquencyRange(Long delinquencyRangeId, JsonCommand command) {
+        DelinquencyRangeData data = dataValidatorRange.validateAndParseUpdate(command);
+        DelinquencyRange delinquencyRange = this.repositoryRange.getReferenceById(delinquencyRangeId);
+        if (delinquencyRange == null) {
+            throw DelinquencyRangeNotFoundException.notFound(delinquencyRangeId);
+        }
+        Map<String, Object> changes = new HashMap<>();
+        delinquencyRange = updateDelinquencyRange(delinquencyRange, data, changes);
+        return new CommandProcessingResultBuilder().withCommandId(command.commandId()).withEntityId(delinquencyRange.getId()).with(changes)
+                .build();
+    }
+
+    @Override
+    public CommandProcessingResult deleteDelinquencyRange(Long delinquencyRangeId, JsonCommand command) {
+        final DelinquencyRange delinquencyRange = repositoryRange.getReferenceById(delinquencyRangeId);
+        repositoryRange.delete(delinquencyRange);
+        return new CommandProcessingResultBuilder().withCommandId(command.commandId()).withEntityId(delinquencyRange.getId()).build();
+    }
+
+    @Override
+    public CommandProcessingResult createDelinquencyBucket(JsonCommand command) {
+        DelinquencyBucketData data = dataValidatorBucket.validateAndParseUpdate(command);
+        Map<String, Object> changes = new HashMap<>();
+        DelinquencyBucket delinquencyBucket = updateOrCreateDelinquencyBucket(data, changes);
+        return new CommandProcessingResultBuilder().withCommandId(command.commandId()).withEntityId(delinquencyBucket.getId()).with(changes)
+                .build();
+    }
+
+    @Override
+    public CommandProcessingResult updateDelinquencyBucket(Long delinquencyBucketId, JsonCommand command) {
+        DelinquencyBucketData data = dataValidatorBucket.validateAndParseUpdate(command);
+        DelinquencyBucket delinquencyBucket = this.repositoryBucket.getReferenceById(delinquencyBucketId);
+        if (delinquencyBucket == null) {
+            throw DelinquencyBucketNotFoundException.notFound(delinquencyBucketId);
+        }
+        Map<String, Object> changes = new HashMap<>();
+        delinquencyBucket = updateDelinquencyBucket(delinquencyBucket, data, changes);
+        return new CommandProcessingResultBuilder().withCommandId(command.commandId()).withEntityId(delinquencyBucket.getId()).with(changes)
+                .build();
+    }
+
+    @Override
+    public CommandProcessingResult deleteDelinquencyBucket(Long delinquencyBucketId, JsonCommand command) {
+        final DelinquencyBucket delinquencyBucket = repositoryBucket.getReferenceById(delinquencyBucketId);
```

Comment:
Shouldn't we handle the case when the method is called for a non-existing delinquencyBucketId; i.e. not found?

Reply:
@josehernandezfintecheandomx EclipseLink will throw immediately "EntityNotFoundException" if the reference does not exist.